Modix.Data/Models/Core/MessageEntity.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.EntityFrameworkCore;
+using Modix.Data.Utilities;
+
+namespace Modix.Data.Models.Core
+{
+    public class MessageEntity
+    {
+        [Required]
+        public ulong Id { get; set; }
+
+        [Required]
+        public ulong GuildId { get; set; }
+
+        [Required]
+        public ulong ChannelId { get; set; }
+
+        [Required]
+        public ulong AuthorId { get; set; }
+
+        [Required]
+        public DateTimeOffset Timestamp { get; set; }
+
+        [OnModelCreating]
+        internal static void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<MessageEntity>()
+                .HasKey(x => x.Id);
+
+            modelBuilder.Entity<MessageEntity>()
+                .Property(x => x.Id)
+                .HasColumnType<ulong>("numeric(20)");
```

Comment:
These should be `.HasConversion<long>()` to match the rest of MODiX.

Reply:
I added this because the migrations generated `<decimal>`. This didn't fix that so I can remove it.